internal/p2p/peermanager.go

```diff
@@ -837,6 +861,7 @@ func (m *PeerManager) Disconnected(ctx context.Context, peerID types.NodeID) {
 		})
 	}
 
+	m.blacklisted[peerID] = time.Now()
```

Comment:
One concern of adding peer to blacklist directly is that this might just be a good peer which happen to be restarted or something. What we can do here is to keep a disconnect frequency for each peer, and only blacklist if a peer constantly disconnected in the recent time window